### ISSUE TITLE
Variable will overflow?

### ISSUE THREAD

**Issue by bsom** [markalanboyd/Audulus-DSP#21]
https://github.com/markalanboyd/Audulus-DSP/blob/main/code/oscillator/test_sine_wave_generator/test_sine_wave_generator.lua

In this example, won't the variable 'phase' eventually overflow?

**Comment by markalanboyd** [comment]
Yes, good catch! When I wrote that I did a back of a napkin calculation in my head and thought, well, it will, but it will take a long time - and it does, if you're only running at 44.1kHz, about 12 hours.

However, if you're supersampling, it would run out a lot faster, so clearly would be a good idea to guard against that.

Please feel free to submit a PR for your idea of how to prevent the overflow - I can duplicate the change across the other oscillators as well.

Thank you!

**Comment by bsom** [comment]
I'm kinda curious what will happen when it does overflow. :-)

**Comment by markalanboyd** [comment]
I think Taylor must have put some protection in there - check this out! Also, I was doing the calculation based on 32-bit maximum before. 

If it's 64-bit, then it could run at 44.1kHz for 6 million years.

At the highest supported sample rate of 192kHz and a x16 resample, it would hypothetically overflow in 6 days - if that's the case, it might be user-choice whether or not to skip adding overflow protection if they want to be really conservative about CPU time.

```lua

acc = 0

function process(frames)
    for i=1,frames do
    	acc = acc + 18446744073709551615
        output[i] = acc
    end
end


```

![Screenshot 2023-04-17 at 17 59 11](https://user-images.githubusercontent.com/10804241/232543342-d0e77dc6-a35f-4320-8dce-483bc84b1a5c.png)

[overflow_test.zip](https://github.com/markalanboyd/Audulus-DSP/files/11252819/overflow_test.zip)

**Comment by markalanboyd** [comment]
Nevermind, haha - got it to flip to `inf` with enough zeros added on there.

If you don't do a PR for the solution to this, I'll do it tomorrow when I have a moment to do it to all of the oscillators at once. Thanks again!
